### ISSUE TITLE
Merging coverage of different tests using the same executable

### ISSUE THREAD

**Issue by mgehre** [SimonKagstrom/kcov#54]
Hi,

thanks again for fixing the shared library coverage. My current issue lies with the merging of coverage from multiple runs of the same executable.
If I run
```
kcov /tmp/kcov app -opt1
kcov /tmp/kcov app -opt2
```
then /tmp/kcov holds only the results of the last invocation of "app".

If I use different directories for each invocation and then use kcov-merge, I seem not to be able to look at individual source file coverage anymore.

Is there a way to aggregate coverage from multiple runs of the same executable (with different options?)

Best wishes,
Matthias

**Comment by SimonKagstrom** [comment]
That's actually something which should just work "out-of-the-box", and I also have regression tests for it in the test suite, which passes.

What kind of application are you running (Python, Bash or a compiled program)?


Does it work with a trivial test, such as a shell script which can be invoked with different parameters:

```sh
#!/bin/bash

if [ $1 -eq 1 ] ; then
   echo "first"
else
   echo "second"
fi
```

kcov-merge only outputs a summary, so individual sources are not present there. Something similar would be possible to add to kcov itself, but I haven't seen it as very important so far.

**Comment by SimonKagstrom** [comment]
Is this in a shared library by the way?

In that case, it might be caused by the library being loaded at a different base address the next time. If this is the case, the bug is caused by kcov storing collected data with actual addresses (which are not meaningful for libraries).

**Comment by mgehre** [comment]
Yes, its in a shared library. I didn't check if it merges the actual exectuable coverage correctly, I was just looking at the library.

**Comment by SimonKagstrom** [comment]
OK. I didn't realize that this was a problem until now. I'll add a regression test and think of a solution.

**Comment by SimonKagstrom** [comment]
Fixed (but lightly tested!) via 96c6eeb.

The implementation remembers the address ranges when shared libraries are added, and stores rangeIndex,offset pairs in the database.


Closing, but feel free to reopen if it turns out to be buggy :-)
